### PR TITLE
Issue #41 Remove ClusterMessage and TransportMessage

### PR DESCRIPTION
**Pull request by antonkharenko**
Main change at this PR related to issue https://github.com/servicefabric/servicefabric/issues/41

Also it contains several minor and cosmetic changes:
* Move logback.xml to testlib module
* Clean up transport exceptions model
* Clean up logging at FailureDetector

### CODE REVIEW

**Comment by ronenhamias** [comment]
+1

File: cluster/src/test/java/io/servicefabric/cluster/ClusterMembershipBuilder.java [hunk 1]
```diff
@@ -179,7 +179,7 @@ private void assertStatus(final ClusterMemberStatus s, TransportEndpoint[] membe
     Predicate<ClusterMember> predicate = new Predicate<ClusterMember>() {
       @Override
       public boolean apply(@Nullable ClusterMember input) {
-        return input.status() == s;
+        return input != null && input.status() == s;
```
Comment (myroslavlisniak) [inline cluster/src/test/java/io/servicefabric/cluster/ClusterMembershipBuilder.java]: is it really nullable?

Comment (antonkharenko) [inline cluster/src/test/java/io/servicefabric/cluster/ClusterMembershipBuilder.java]: hm, I think no

File: transport/src/main/java/io/servicefabric/transport/ConnectorHandshakeChannelHandler.java [hunk 2]
```diff
@@ -20,12 +18,12 @@
...
- * (rest inbound messages unsupported and results in {@link TransportBrokenException}). On outbound may enqueue
- * messages. On 'channel active' starting handshake process.
+ * Duplex handler. On inbound recognizes only handshake message
+ * {@link TransportHandshakeData#Q_TRANSPORT_HANDSHAKE_SYNC_ACK} (rest inbound messages unsupported and results in
+ * {@link TransportBrokenException}). On outbound may enqueue messages. On 'channel active' starting handshake process.
  * <p/>
- * <b>NOTE:</b> this handler is not shareable (see {@link #sendMailbox}, {@link #handshakeTimeout}); and should always run
- * in different executor than io-thread.
+ * <b>NOTE:</b> this handler is not shareable (see {@link #sendMailbox}, {@link #handshakeTimeout}); and should always
+ * run in different executor than io-thread.
```
Comment (myroslavlisniak) [inline transport/src/main/java/io/servicefabric/transport/ConnectorHandshakeChannelHandler.java]: For future: we just prove that switch between executors kills performance, so need to think again is it really need to run in different executor. 
@artem-v can you look?

**Comment by myroslavlisniak** [comment]
+1

**Comment by artem-v** [comment]
+1

**Pull request merged**